cli/gardener_ci/protecode_cli.py

```diff
@@ -75,17 +81,19 @@ def scan_without_notification(
         cvss_version=CVSSVersion(cvss_version),
     )
 
-    logger.info('preparing license report for protecode results')
-    concourse.steps.scan_container_images.print_license_report(license_report)
-    updated_license_report = list(
-        concourse.steps.scan_container_images.determine_rejected_licenses(
-            license_report,
-            allowed_licenses,
-            prohibited_licenses,
+    if not no_license_report:
+        logger.info('preparing license report for protecode results')
+        concourse.steps.scan_container_images.print_license_report(license_report)
+        updated_license_report = list(
+            concourse.steps.scan_container_images.determine_rejected_licenses(
+                license_report,
+                allowed_licenses,
+                prohibited_licenses,
+            )
         )
-    )
 
     logger.info(f'{len(results_above_threshold)=}; {results_above_threshold=}')
     logger.info(f'{len(results_below_threshold)=}; {results_below_threshold=}')
-    logger.info(f'{len(updated_license_report)=}; {updated_license_report=}')
+    if not no_license_report:
+        logger.info(f'{len(updated_license_report)=}; {updated_license_report=}')
```

Comment:
From reading the code: Shouldn't this be called `rejected_licenses_report` or something similar? Also, if that's really only the rejected licenses that are found, we could roll it in the first `if not no_license_report` block

Reply:
change head-lines of logs to make it more transparent, only one if block, keep the command-line option as is to keep it short.